src/main/java/elements/CheckBoxImpl.java

```diff
@@ -0,0 +1,37 @@
+package elements;
+
+
+import org.openqa.selenium.ElementNotInteractableException;
+import org.openqa.selenium.ElementNotVisibleException;
+import org.openqa.selenium.WebElement;
+import util.ScrollUtils;
+import util.WaitUtils;
+
+
+/***
+ * Wrapper class with Checkbox functionalities
+ */
+public class CheckBoxImpl extends ElementImpl implements Checkbox {
+
+    public CheckBoxImpl(WebElement element) {
+        super(element);
+    }
+
+    @Override
+    public void clickCheckbox(boolean isChecked) {
```

Comment:
Please update this method logic. You need to get currect checkbox state, compare to expected and if it's the same - don't anything, if it needs to be changed - click

Reply:
done